plugins/pipelines/app/decorators/stage_decorator.rb

```diff
@@ -19,6 +19,12 @@ def next_stages
     Stage.where(id: next_stage_ids).to_a
   end
 
+  def previous_stages
+    @previous_stages ||= project.stages.to_a.select do |stage|
+      stage.next_stage_ids.map(&:to_i).include? id
```

Comment:
@grosser is there some way to force `next_stage_ids` to be an array of ints?  It's pretty unintuitive that the method returns an array of strings.